photonField.py

```diff
@@ -546,6 +546,40 @@ def load_density(self, fname):
 
         return energy, photonDensity
 
+class EBL_Finke22(EBL):
+    """ IRB model from FInke et al. 2022 """
+    
+    def __init__(self):
+        super(EBL_Finke22, self).__init__()
+        self.name = 'IRB_Finke22'
+        self.info = 'Cosmic infrared and optical background radiation model of Finke et al. 2022 (Model A).'
+        self.files = os.path.join(datadir, 'EBL_Finke_2022', 'z%.2f.dat')
+        self.redshift = np.arange(0, 7.02 , 0.02)
+        self.energy = []
+        self.photonDensity = []
+        self.load_data()
+    
+    def load_data(self):
+        tmp_dict = {}
+        
+        for i, z in enumerate(self.redshift):
+            data = pd.read_csv(self.files % z, sep="\s+", header=None)
+            # data[0]: photon energy [eV]
+            # data[1]: comoving energy density [erg/cm^3]
+            eps = data[0].to_numpy()
+            n = data[1].to_numpy()
+            
+            self.data[z] = (eps * eV) , (n * erg * 1e6 / (eps * eV)**2)   # [J] , dn/deps [1/m^3/J]
+            
+            # adding the energies only once, cause they are the same in every file
+            if i == 0:
+                tmp_dict['energy'] = eps
+            tmp_dict[str(z)] = n
```

Comment:
You can directly append to `self.photonDensity`. This would save a lot of overhead in storing the data.

Reply:
I updated that part a bit. However, the tmp construction seems useful to me. The data is differently structured in the raw files compared to our storage. Therefore, when loading the files it somehow has to be transposed before stored into the photonField variable; or a more complex loading would be needed.